python/ray/serve/_private/http_state.py

```diff
@@ -230,6 +230,26 @@ def shutdown(self):
         self._shutting_down = True
         ray.kill(self.actor_handle, no_restart=True)
 
+    def is_ready_for_shutdown(self) -> bool:
+        """Return whether the HTTP proxy actor is shutdown.
+
+        For an HTTP proxy actor to be considered shutdown, it must be marked as
+        _shutting_down and the actor must be dead. If the actor is dead, the health
+        check will return RayActorError.
+        """
+        if not self._shutting_down:
+            return False
+
+        try:
+            finished, _ = ray.wait(
+                [self._actor_handle.check_health.remote()], timeout=0
+            )
+            ray.get(finished[0])
```

Comment:
this block looks unsafe -- can't the ref not be returned in `finished` in which case we'll get an index error on the `finished[0]`?

Reply:
I would suggest something like the following:
```suggestion
            ray.get(self._actor_handle.check_health.remote(), timeout=0.001)
```

then catch both timeout and ray actor errors. timeout means not ready, actor error means it's shut down

Reply:
Oh cool, let me change real quick!